partial-update/src/main/java/com/azure/autorest/partialupdate/util/PartialUpdateHandler.java

```diff
@@ -474,6 +490,11 @@ private static boolean isMembersCorresponding(BodyDeclaration<?> member1, BodyDe
         if (member1.isCallableDeclaration() && member2.isCallableDeclaration()) {
             // compare signature
             return member1.asCallableDeclaration().getSignature().equals(member2.asCallableDeclaration().getSignature());
+        } else if (member1 instanceof InitializerDeclaration && member2 instanceof InitializerDeclaration
+                && ((InitializerDeclaration) member1).isStatic() && ((InitializerDeclaration) member2).isStatic()) {
```

Comment:
so assumption is generated static block is already same with existing static block? would add this into comment?

Reply:
No such assumption.

It is "always replace" (same logic, if method on both side is `@Generated`).

Reply:
Sorry for using the wrong word 'same',  'corresponding' is better. both `@Generated` and method/code block is corresponding, then we will replace.

Reply:
Added the comment to Javadoc of the method as well.